docs/integrations/consul/k8s.md

```diff
@@ -0,0 +1,313 @@
+  Next, we'll install the ngrok Ingress Controller into our cluster. We want the controller pods to be in the Consul service mesh in order to proxy traffic to our other services. We'll use pod annotations to enable the Consul Connect sidecar injector and allow outbound traffic to use the Consul mesh. Consul documents to set these 2 annotations in the [Configure Ingress Controllers for Consul on Kubernetes](https://developer.hashicorp.com/consul/docs/k8s/connect/ingress-controllers) doc.
+
+    ```yaml
+    # This annotation is required to enable the Consul Connect sidecar injector
+    consul.hashicorp.com/connect-inject: "true"
+    # This is the CIDR of your Kubernetes API: `kubectl get svc kubernetes --output jsonpath='{.spec.clusterIP}'
+    consul.hashicorp.com/transparent-proxy-exclude-outbound-cidrs: "10.96.0.1/32"
+    ```
+
+  ```bash
+  helm install ngrok-ingress-controller ngrok/kubernetes-ingress-controller \
+    --namespace consul \
+    --set fullnameOverride=ngrok-ingress-controller \
+    --set credentials.apiKey=$NGROK_API_KEY \
+    --set credentials.authtoken=$NGROK_AUTHTOKEN \
+    --set-string podAnnotations.consul\\.hashicorp\\.com/connect-inject=true \
+    --set podAnnotations."consul\.hashicorp\.com/transparent-proxy-exclude-outbound-cidrs"="YOUR KUBERNETES API CIDR"
+  ```
+
+:::note
+- Hashicorp's docs also mention the annotation `consul.hashicorp.com/transparent-proxy-exclude-inbound-ports`. This is not applicable to the ngrok Ingress Controller as we create an outbound connection for Ingress rather than exposing ports.
+- The `--set-string` flag allows the pod annotation to escape the `.` character in the annotation name while ensuring the value `true` is a boolean and not a string.
+- In a production environment, or anywhere you wish to use Infrastructure as Code and source control your Helm configurations, you can setup your credentials following this [guide](https://github.com/ngrok/kubernetes-ingress-controller/blob/main/docs/deployment-guide/credentials.md).
+:::
+
+5. Verify the ngrok Ingress Controller is installed and all its pods are healthy
+  ```bash
+  kubectl get pods -l 'app.kubernetes.io/name=kubernetes-ingress-controller' -n consul
+  NAME                                                READY   STATUS    RESTARTS      AGE
+  ngrok-ingress-controller-manager-5b796c88f7-k7v6z   2/2     Running   1 (64s ago)   67s
+  ```
+
+
+## **Step 3**: Install a Sample Application {#install-a-sample-application}
+
+Now let's install a sample application to try out our service mesh and Ingress Controller combination. We'll use the [HashiCups Demo Application](https://github.com/hashicorp-demoapp) Hashicorp uses for demos and guides such as in their [Getting Started with Consul Service Mesh for Kubernetes](https://developer.hashicorp.com/consul/tutorials/kubernetes-features/service-mesh-deploy) guide. This application is a simple e-commerce application that allows users to order coffee cups.
```

Comment:
We'll use the [HashiCups Demo Application](https://github.com/hashicorp-demoapp) Hashicorp uses 

_HashiCorp_ uses